### ISSUE TITLE
Making Unicode supported LaTeX template as the default

### ISSUE THREAD

**Issue by thammegowda** [acl-org/acl-style-files#7]
We have been using PdfLatex compiler/engine as the default, but as we know it isn't Unicode (non-Latin) friendly.
Though the instructions suggest using XeLaTeX,  the generated PDF looks different in many ways than PdfLatex's.
For example (left: PdfLatex, right: XeLatex): Look at the nuances in fonts,  section headings aren't as bold as PdfTex's in the left.  I believe the font weight isn't exactly the same.

<img width="1673" alt="Screen Shot 2022-01-04 at 2 55 02 PM" src="https://user-images.githubusercontent.com/1865964/148135515-e295b22b-1fbd-4c3f-ba02-c0d54326e56d.png">

My request/suggestion:
Move towards Unicode supported template as a way of encouraging NLP in non-Latin languages. 
Researchers working on non-Latin languages should also be able to paste qualitative examples (without some non-vector images), right? So, how about making Unicode supported template (i.e XeLatex) as the default? 
 
If any one interested in testing unicode support of latex templates, here is a file having UDHR titles in hundreds of languages:
[udhr-title.txt](https://github.com/acl-org/acl-style-files/files/7810805/udhr-title.txt)

Thanks,

**Comment by davidweichiang** [comment]
This all sounds great. But we need to set it up so that it looks the same either way.

**Comment by davidweichiang** [comment]
XeLaTeX has a major disadvantage, which is that arXiv does not support it. So I don't think it can be made the default (yet). But I definitely agree with making it an option.

@thammegowda In your example, the one on the right is set in Computer Modern, not Times Roman. So something is wrong with the font setup.

**Comment by thammegowda** [comment]
The modifications I did to add some Unicode text was
1. enable babel 
```latex
\usepackage[english]{babel} % English as the main language
\babelprovide[import]{hindi}
\babelprovide[import]{arabic}
\babelprovide[import]{kannada}
\babelfont[*devanagari]{rm}{Lohit Devanagari}
\babelfont[*arabic]{rm}{Noto Sans Arabic}
```

2. Paste some Arabic and Hindi text 
```latex
Hindi: \foreignlanguage{hindi}{मानव अधिकारों की सार्वभौम घोषणा} Arabic: \foreignlanguage{arabic}{الإعلان العالمي لحقوق الإنسان
```

3. And switch compiler to XeLaTex, since PdfTex could not compile it.
 Also, I had to comment out `\pdfoutput=1`  for XeLaTex
 
 I didn't explicitly modify fonts for English/Latin. Is `babel` import messing up default fonts for English? Sorry, I am not a *TeX pro.   Here is my overleaf project for reference  https://www.overleaf.com/project/61d4c64cbc3e72789d2de4bc

**Comment by mbollmann** [comment]
Well, I would say arXiv has a major disadvantage in that it doesn't support XeLaTeX/LuaLaTeX, but I can see how we should make sure to support it ;)

@thammegowda The default font is Computer Modern, to get the correct font for the current *ACL template, both `\usepackage{times}` and `\usepackage[T1]{fontenc}` are important.

**Comment by thammegowda** [comment]
@mbollmann I agree,  and I hope arXiv realizes this shortcoming and makes an update. 

Also,  I have these two lines 
```latex
\usepackage{times}
\usepackage[T1]{fontenc}
```

I didn't remove these two, but is XeLaTex using Computer Modern? That's surprising!

**Comment by mbollmann** [comment]
@thammegowda Ah, maybe it is overwritten by something else in your preamble then. I can't access your Overleaf project, it's restricted. Try to move the "times" import further down maybe?

**Comment by mbollmann** [comment]
@thammegowda Not an expert with Babel, but I think as soon as you use a `\babelfont`, you need to define an explicit Latin font as well. I haven't found a way to get the _exact_ same font as LaTeX's `ptm` family (which is what "times" uses), but if you add

```latex
\babelfont{rm}{TeX Gyre Termes}
```

before you load the other, language-specific fonts, you get something virtually indistinguishable from it.

**Comment by thammegowda** [comment]
That works! Thanks.

**Comment by venkatasg** [comment]
I was looking into whether there were efforts to move away from `pdflatex` to be more Unicode friendly. I have 2 suggestions, and can help with the migration in these respects:

- As an engineer LuaLaTeX is probably better than XeLaTeX for the reasons [discussed here](https://tex.stackexchange.com/a/126216/120524).
- moving away from the Times font to [Charter](https://practicaltypography.com/charter.html) provided by packages like [XCharter](https://www.ctan.org/pkg/xcharter). We have many good free fonts today, like Charter, there's no need to still be using Times. Furthermore, [Charis SIL](https://software.sil.org/charis/) was based on Charter and will play well with it. Supporting IPA, and wide variety of character sets natively ought to be a priority, especially considering the object of our study.

Further decisions probably need to be made about sans-serif and monospaced fonts, but none that can't be solved with some research.

**Comment by LoicGrobol** [comment]
Another consideration is that [tagged pdf support](https://www.texdev.net/2024/11/05/engine-news-from-the-latex-project) and its massive advantages for accessibility will probably only come fully for LuaLaTeX. ArXiv has announced plans to support it, given their pace probably not in the near future, but we could add instructions for it in the template (like commented-out lines for using fontspec etc.), or even making it the default and keeping instructions for porting to ArXiv (which typically already requires extra steps anyway, like stripping comments, generating the bbl file, etc.)

**Comment by LSinev** [comment]
Even commenting out of (pdf|xe|lua)latex is not necessary, because with `iftex` package new condition can be used, like:
```
\newif\ifxetexorluatex   % http://tex.stackexchange.com/a/47579
\ifxetex
    \xetexorluatextrue
\else
    \ifluatex
        \xetexorluatextrue
    \else
        \xetexorluatexfalse
    \fi
\fi
```
and then `\ifxetexorluatex` can be used to load different packages for pdflatex and for lualatex|xelatex (https://github.com/AndreyAkinshin/Russian-Phd-LaTeX-Dissertation-Template/blob/master/common/packages.tex#L64-L101 example for Russian).

I can help with such PR if some Hindi/Arabic/Japanese/Korean (and other hard for font encoding support cases) will be provided.

**Comment by danielgildea** [comment]
I added an example file that compiles with either luaLaTeX or XeLaTeX.

I don't think it makes sense to change the default until arxiv supports luaLaTeX and/or it becomes more standard generally.  Providing commands for people to comment/uncomment seems more complicated than just providing a second template file.

**Comment by davidweichiang** [comment]
I don't see an issue with a single style file that uses conditionals to work under all three engines, and think that will be more maintainable than two separate style files.

I think people are very used to the appearance of the ACL style and a font change from Times to something else probably wouldn't go over well.

**Comment by LoicGrobol** [comment]
I agree that conditions seem like a better idea. I'm not overly fond of Times-like fonts in general, but TeX Gyre Termes is pretty indistinguishable, while still having better character support.

And also, when using LuaLaTeX, the new font backends (fontspec/harfbuzz) makes it quite easy to create composite font for missing characters or blocks (even color emoji!), I've been doing it for years now and can very easily add it to the template.

**Comment by davidweichiang** [comment]
I'd be okay with a switch to TeX Gyre Termes. However, there would need to be coordination with https://github.com/acl-org/aclpubcheck .

I didn't understand @LoicGrobol's proposal about composite fonts. What are you proposing to add to the template?

For specific language support, I would lean against baking anything into `acl.sty`. At least for Chinese, there are several common methods and I don't think people would be happy with being told which one to use.

My recollection is that there is one language baked into acl.sty (and commented out?) and it's Vietnamese. This was because in the Anthology, Vietnamese was by far the most common non-Latin-1 language we saw in BibTeX entries. So authors may need Vietnamese support even if there's no Vietnamese in the body of the paper.

**Comment by LoicGrobol** [comment]
@davidweichiang Oh I was thinking of something to the effect of https://tex.stackexchange.com/q/619555/8547, but probably not right at this moment

**Comment by danielgildea** [comment]
Hi,

To be precise, there is still only one style file, acl.sty, and no changes were necessary to make it work with luaLaTeX and XeLaTeX.

There are now two template files showing people how to use acl.sty.  They could be combined with conditionals, but 

1) they don't share that much content, and

2) it would be awkward to still include \pdfoutput=1 in the first five lines for arxiv submissions.

**Comment by davidweichiang** [comment]
Oh, agreed, for the example (template) file, two separate files is great.
